### DISCUSSION TITLE
Allowed Extensions option

### DISCUSSION THREAD

**Discussion by phxn** [sabnzbd/sabnzbd#2340] [Ideas]
It would be nice if there was an option such as "Allowed Extensions".

So if you're only interested in *ahem* copyright free video, you could set the field to allow only "mkv, mp4" and all files belonging to other extension would be auto deleted.

Why am I suggesting this? Well, at the moment you have to filter every single extension known to man to avoid getting stuff you don't want. This way, you'd only need to list the extensions you want to keep, which for most people are very few.

Whatya think?

**Comment by sanderjo** [comment]
> and all files belonging to other extension would be auto deleted.

So I guess you're referring to config/switches/#cleanup_list which says:

> Cleanup List
> List of file extensions that should be deleted after download.
> For example: nfo or nfo, sfv
> 

(and you aren't referring to Unwanted Extension, which blocks the download)

And so you want a Keep List instead of Cleanup/Delete List? And into that Keep List you would put "iso,mkv, mp4".

Correct?

**Reply by phxn** [reply]
Yep, exactly.

At the moment you have to filter everything you don't want, I am suggesting an option to filter only the things you want to keep. Any other extensions not matching your "Allowed Extensions" will be auto deleted.

It's far easier to choose which type of files to keep, rather than adding every possible extension that could slip through the net if you haven't added it to the Cleanup List.

**Reply by thezoggy** [reply]
i would beg to differ, but of course everyone is different. its hard to know what subtitle format someone might use or what random files you may want if your grabbing apps/games..   generally people just have solved this with using a post processing script so they can dictate per category what stuff to remove or keep.

**Reply by phxn** [reply]
People would just whitelist those though? If you want to keep all possible subtitles, you can add them.

This wouldn't be mandatory. You could skip the feature all together if you didn't want to use it--Just like the Cleanup List.

**Comment by Safihre** [comment]
Proposal:

![image](https://user-images.githubusercontent.com/5703454/201686953-1b23b611-33d9-4ae2-9a57-ecf0e1ebbcd6.png)

**Reply by phxn** [reply]
Looks great!

**Reply by thezoggy** [reply]
I could see people causing themselves pain by not realizing the change in action if that dropdown got changed (mouse scroll wheel). So I'd say we prob should show a warning if the action is changed so people know they prob need to revisit that cleanup list (or we just wipe the cleanup list on change.. but that prob would also cause some grumbles)

**Reply by Safihre** [reply]
I'm surprised @thezoggy that you expect so many problems with this one. Is it so much different than any other drop down setting that could be accidentally changed? 
I have seen quite some users that have a giant list in Cleanup just to get rid of everything nonsense so I like this feature quite a lot.

**Reply by thezoggy** [reply]
to be clear, I have no problem with the feature. ideally it should be expanded so its per cat (as your cleanup list for movies is prob a bit different than apps) for example.   

my only caveat is people going from one action to the other, and the gotcha there.
It shouldnt' be a big thing if someone is actively changing it and not realizing what its going to do, but mainly it was those that accidentally did it. but all good, if it comes up during support questions then we can worry about guardrails.

**Reply by phxn** [reply]
Having it as an option per category would be nice too. Although I suspect a lot of work?

**Comment by phxn** [comment]
Hello, was there any final decision or any updates on this? Would love to see it happen :)

**Reply by Safihre** [reply]
It seems like a fine feature, someone just needs to take the time to build it 😊

**Reply by phxn** [reply]
Every time I see an update I get excited :)
